torchvision/prototype/datasets/video_utils.py

```diff
@@ -0,0 +1,192 @@
+from typing import Any, Callable, Dict, List, Optional, Tuple, Mapping, Union, Iterator
+import random
+import av
+import numpy as np
+import torch
+from torchdata.datapipes.iter import IterDataPipe
+from torchvision.io import video, _video_opt
+
+class AVKeyframeReader(IterDataPipe[Dict[str, Any]]):
+    def __init__(self, video_dp: IterDataPipe[Dict[str, Any]], transform=None) -> None:
+        """TorchData Iterdatapype that takes in video datapipe
+        and yields all keyframes in a video
+
+        Args:
+            video_dp (IterDataPipe[Dict[str, Any]]): Video dataset IterDataPipe
+            transform ([type], optional): Transform to be applied to every frame. Defaults to None.
+        """
+        self.datapipe = video_dp
+        self.transform = transform
+
+    def __iter__(self) -> Iterator[Dict[str, Any]]:
+        for video_d in self.datapipe:
+            buffer = video_d["file"]
+            with av.open(buffer, metadata_errors="ignore") as container:
+                stream = container.streams.video[0]
+                stream.codec_context.skip_frame = 'NONKEY'
+                for frame in container.decode(stream):
+                    img = frame.to_image()
+                    if self.transform is not None:
+                        img = self.transform(img)
+                    yield {
```

Comment:
We should make sure that we include all the information that we had previously and only replace `"file"` with new data. The easiest way to do that is to `pop` the key and then use the old data as base for the new dictionary.

```suggestion
                    yield dict(video_d, frame=img, ...)
```